src/language-server.ts

```diff
@@ -4,122 +4,152 @@ import { ExtensionContext, ProgressLocation, Uri, window } from "vscode";
 import { LanguageClient, LanguageClientOptions, ServerOptions } from "vscode-languageclient/node";
 import * as unzipper from 'unzipper';
 import { join } from "path";
-import * as InstallationManifest from "./installation-manifest";
-
-export async function start(releasePath: string): Promise<void> {
-	const outputChannel = window.createOutputChannel("Lexical");
-
-	const startScriptPath = join(releasePath, "start_lexical.sh");
-
-	const serverOptions: ServerOptions = {
-		command: startScriptPath,
-	};
-
-	const clientOptions: LanguageClientOptions = {
-		outputChannel,
-		// Register the server for Elixir documents
-		// the client will iterate through this list and chose the first matching element
-		documentSelector: [
-			{ language: "elixir", scheme: "file" },
-			{ language: "elixir", scheme: "untitled" },
-			{ language: "eex", scheme: "file" },
-			{ language: "eex", scheme: "untitled" },
-			{ language: "html-eex", scheme: "file" },
-			{ language: "html-eex", scheme: "untitled" },
-			{ language: "phoenix-heex", scheme: "file" },
-			{ language: "phoenix-heex", scheme: "untitled" },
-			{ language: "surface", scheme: "file" },
-			{ language: "surface", scheme: "untitled" },
-		],
-	};
-
-	const client = new LanguageClient(
-		"lexical",
-		"Lexical",
-		serverOptions,
-		clientOptions
-	);
-
-  outputChannel.appendLine(`Starting lexical release in "${releasePath}"`);
-
-	await client
-		.start()
-		.then(() => {
-			console.log("Started Lexical");
-		})
-		.catch((reason) => {
-			window.showWarningMessage(`Failed to start Lexical: ${reason}`);
-		});
-}
+import InstallationManifest from "./installation-manifest";
+import Github from "./github";
+import Release from './release';
+
+namespace LanguageServer {
```

Comment:
Why use a namespace?

Reply:
Namespaces allow us to export types and functions under a common identifier. Without namespaces, we have to either:
- use `export default` to export an object containing all functions, and then export the types by name, requiring that all types to have sense on their own (eg. `LexicalRelease` instead `Lexical.Release`). This causes repetition and forces users to use two different import styles
- export everything by name and use `import * as ModuleName` to import the whole scope. The big issue with this is that it provides poor code completion (at least in vscode) because the module doesn't have a name, so you have to import individual functions/types and then redo the import manually.

Honestly I haven't seen this pattern used much in the ecosystem, but I think it's closer to what Elixir devs are used to, which was my goal in order to facilitate contributions from the Elixir community.

Reply:
I have no objection to using a small amount of namespace